File: src/types/ai.ts
// ── AI / Chat types ──────────────────────────────────────────────────────────
export type AIDataType =
  | 'financial_data'
  | 'calculation'
  | 'retrieved_information'
  | 'insight';

export interface AIDataBlock {
  type: AIDataType;
  title: string;
  values?: Record<string, unknown>;
}

export interface AISource {
  title: string;
  url: string;
  publisher?: string;
}

export interface AIChatRequest {
  message: string;
  conversationId?: string;
  context?: {
    page?: string;
    transactionIds?: string[];
    loanId?: string;
    goalId?: string;
  };
}

export interface AIChatResponse {
  message: string;
  intent?: string;
  data?: AIDataBlock[];
  sources?: AISource[];
  calculation?: {
    type: string;
    result: unknown;
  };
  confidence?: number;
  disclaimer?: string;
}

export interface ChatMessage {
  id: string;
  role: 'user' | 'assistant';
  content: string;
  timestamp: string;
  data?: AIDataBlock[];
  sources?: AISource[];
  disclaimer?: string;
}

export interface Conversation {
  id: string;
  title: string;
  createdAt: string;
  updatedAt: string;
  messages: ChatMessage[];
}

File: backend/app/schemas/ai.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any


class AIChatRequest(BaseModel):
    message: str
    conversationId: Optional[str] = None
    context: Optional[Dict[str, Any]] = None


class AIDataBlock(BaseModel):
    type: str
    title: str
    values: Optional[Dict[str, Any]] = None


class AISource(BaseModel):
    title: str
    url: str
    publisher: Optional[str] = None


class AIChatResponse(BaseModel):
    message: str
    intent: Optional[str] = None
    data: Optional[List[AIDataBlock]] = None
    sources: Optional[List[AISource]] = None
    disclaimer: Optional[str] = None


class ConversationOut(BaseModel):
    id: str
    title: str
    createdAt: str
    updatedAt: str

    model_config = {"from_attributes": True}
